src/adapters/book-adapters.ts

```diff
@@ -0,0 +1,245 @@
+import { setCookie, getCookie }from'../util/Authentication'
+export const fetchAPIData = async (url: string, method: string, body:object | undefined , headers?:object) => {
+    const cookie = getCookie('shelf-share-session')as { token: string }
+    const token = cookie?.token
+    const response = await fetch(url , {
+        method,
+        body: JSON.stringify(body),
+        headers: {
+            Authorization: `Bearer ${token}`,
+            'Content-Type': 'application/json',
+            ...headers,
+        },
+    })
+    
+    if(response.status.toString().startsWith('4')||response.status.toString().startsWith('5')){
+      console.log("Error", response.status)
+    }
+    else{
+      const data = await response.json();  
+      return data; 
+    }   
```

Comment:
Lara, We suppose to import this fetch wrapper from fetch.ts!

Reply:
Ok,  I could import it by adding undefined on the body: object on fetch.ts